code/datums/diseases/advance/symptoms/wizarditis.dm

```diff
@@ -100,10 +100,38 @@
 
 
 /datum/symptom/wizarditis/proc/teleport(datum/disease/advance/A)
-	var/turf/L = get_safe_random_station_turfs()
-	A.affected_mob.say("SCYAR NILA!")
-	do_teleport(A.affected_mob, L, channel = TELEPORT_CHANNEL_MAGIC)
-	playsound(get_turf(A.affected_mob), 'sound/weapons/zapbang.ogg', 50,1)
+	var/list/theareas = get_areas_in_range(80, A.affected_mob)
+	for(var/area/space/S in theareas)
+		theareas -= S
+
+	if(!theareas||!theareas.len)
+		return
+
+	var/area/thearea = pick(theareas)
+
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(T.get_virtual_z_level() != A.affected_mob.get_virtual_z_level())
+			continue
+		if(isgroundlessturf(T))
+			continue
+		if(!T.density)
+			var/clear = 1
+			for(var/obj/O in T)
+				if(O.density)
+					clear = 0
+					break
+			if(clear)
+				L+=T
+
+	if(!L)
+		return
+
+	if(do_teleport(A.affected_mob, pick(L), channel = TELEPORT_CHANNEL_MAGIC, no_effects = TRUE))
+		if(A.affected_mob.stat != DEAD)
+			A.affected_mob.say("SCYAR NILA [uppertext(thearea.name)]!", forced = "wizarditis")
+		playsound(get_turf(A.affected_mob), 'sound/weapons/zapbang.ogg', 50,1)
+	return
```

Comment:
Rather than being duplicated this should just be made into a proc and used in both this and the wizard teleport.

Reply:
will do